### ISSUE TITLE
Add CDR VI benchmark with ARBIS, multifidelity GP, and Newton ablations

### ISSUE THREAD

**Issue by eparish1** [Pressio/rom-tools-and-workflows#393]
## Summary

Add a YAML-driven CDR variational-inference benchmark that mirrors the analytic sine benchmark structure while exercising the ROM-tools convection-diffusion-reaction model. The benchmark should isolate the effects of Adam vs Newton optimization, ARBIS sample reuse, multifidelity GP control variates, and score-function entropy treatment.

Use a **leave-one-out (LOO) baseline for every score-function estimator path** so comparisons isolate optimizer, reuse, multifidelity, and entropy effects rather than baseline choice.

The benchmark should follow the same configuration pattern as the analytic sine benchmark in `eparish1/mfvi-paper#3` / PR #4: one YAML for the primary method ablation and a second YAML for the sample-size sweep, with thin drivers that consume those configurations rather than hard-coding scientific settings.

## Method families and production runs

There are five method families, with the MFVI-Newton family evaluated under two entropy strategies. This gives **six production runs** for the primary comparison:

1. **BBVI (Adam)**
   - Adam optimizer
   - no sample reuse
   - single fidelity
   - LOO baseline

2. **ARBIS**
   - Adam optimizer
   - importance-sampling sample reuse enabled
   - single fidelity
   - LOO baseline

3. **ARBIS + MF**
   - Adam optimizer
   - importance-sampling sample reuse enabled
   - multifidelity BBVI
   - automatic GP surrogate as the low-fidelity model
   - LOO baseline

4. **BBVI (Newton)**
   - Newton optimizer
   - no sample reuse
   - single fidelity
   - LOO baseline

5a. **MF-BBVI (Newton, analytic entropy)**
   - Newton optimizer
   - no sample reuse
   - multifidelity BBVI
   - automatic GP surrogate as the low-fidelity model
   - LOO baseline
   - `score_function_entropy_strategy: analytic`

5b. **MF-BBVI (Newton, joint entropy)**
   - same setup as 5a
   - `score_function_entropy_strategy: joint`

For the entropy ablation, **the two MFVI-Newton runs must differ only in entropy treatment**. Keep the same FOM samples/batch size, ROM-extra samples, GP settings, initialization, Newton/Hessian settings, line search, random seeds, bounds, observations, and FOM budget.

The ROM-tools semantics should be used directly:

- `analytic`: estimate the log-joint score contribution and add the Gaussian entropy gradient analytically;
- `joint`: keep `-log q` inside the sampled ELBO integrand and estimate the full score-function contribution stochastically.

Do not add a coarse-grid CDR model as a low-fidelity model. All multifidelity runs use the same on-the-fly automatic GP ROM built from FOM samples.

## CDR inverse problem

Use the reusable steady CDR example model and the full four-parameter bounded domain already used by the ROM-tools CDR UQ example:

- `bmag` in `[0.5, 1.5]`
- `theta` in `[pi/6, pi/3]`
- `nu` in `[0.02, 0.08]`
- `sigma` in `[0.1, 0.6]`

These bounds are part of the benchmark definition. Store them explicitly in YAML and enforce the same bounded-parameter handling for every run. The synthetic truth must lie strictly inside this domain and should be off-center so the initial variational distribution is not centered on the generating parameters.

Use a common physical observation definition for all methods. A reasonable choice is right-boundary flux evaluated/interpolated at fixed physical sensor locations.

### Observation noise

Use additive independent Gaussian noise with a **5% noise-to-signal ratio based on the global RMS of the clean observation vector**. For

`y_clean = G(theta_truth)`

with `N_obs` observations, define

`sigma_obs = 0.05 * ||y_clean||_2 / sqrt(N_obs)`

and generate

`eta ~ N(0, sigma_obs^2 I)`

`y_obs = y_clean + eta`.

Do not use pointwise relative noise such as `sigma_i = 0.05*|y_i|`.

Generate one noisy observation vector with a fixed observation-noise seed. Every method and every algorithmic random-seed replicate must infer against that same fixed dataset. Save `noise_to_signal_ratio: 0.05`, observation seed, `sigma_obs`, and preferably the realized `||eta||_2/||y_clean||_2` ratio.

## Automatic GP low-fidelity model

For `ARBIS + MF` and both MF-BBVI Newton entropy runs:

- use `mf_vi_with_auto_rom(..., rom_type="gp")` or the equivalent current API;
- normalize parameters and targets;
- use the existing POD treatment for vector-valued QoIs when needed;
- construct/rebuild the GP only from FOM data generated by the workflow;
- do not use a separate offline training set;
- count every FOM evaluation used to train/rebuild the GP against the method's FOM cost.

The GP configuration and rebuild policy must be identical across all multifidelity runs.

## YAML-driven configuration

Provide two YAML files in the CDR benchmark directory:

- `configs/method_ablation.yaml` for the primary six-run comparison;
- `configs/sample_size_sweep.yaml` for the sample-size sweep.

Drivers should accept `--config`, defaulting to the corresponding YAML file. Scientific and algorithmic settings belong in YAML rather than duplicated constants in Python.

The YAML should own at minimum:

- cumulative FOM budget;
- FOM sample size;
- ROM-extra sample size;
- random seed / production seed set;
- maximum iterations;
- CDR truth, prior, explicit parameter bounds, observation definition, 5% RMS noise specification, and observation seed;
- Adam settings;
- Newton metric, Hessian, regularization, and line-search settings;
- ARBIS/sample-reuse settings and diagnostics;
- GP-ROM training-history/rebuild settings;
- normalization settings;
- bounded-parameter handling and run-directory behavior;
- score-function entropy strategy for each relevant run.

Keep method-specific sections such as `bbvi`, `newton`, `sample_reuse`, `rom`, and `common`. The entropy ablation should be represented cleanly in YAML rather than by hard-coded branching. For example, the MFVI-Newton section may expose

```yaml
entropy_strategies: [analytic, joint]
```

or equivalent explicit run definitions.

Set `baseline_method: loo` for all applicable estimator paths and validate it in the benchmark metadata.

Copy the resolved configuration into each output directory.

## Sample-size sweep

Add a second YAML-driven sweep driver analogous to the sine benchmark's `run_fom_sample_size_ablation.py`.

Default sweep axes:

```yaml
sweep:
  fom_sample_sizes: [4, 8, 16, 32]
  mf_rom_extra_sample_sizes: [64, 128]
```

For each `fom_sample_size`, run:

- `BBVI (Adam)`;
- `ARBIS`;
- `BBVI (Newton)`.

For each Cartesian pair of `fom_sample_size` and `mf_rom_extra_sample_size`, run:

- `ARBIS + MF`;
- `MF-BBVI (Newton, analytic entropy)`;
- `MF-BBVI (Newton, joint entropy)`.

A sweep case should copy the base configuration and override only swept sample-size fields. Optimizer settings, LOO baseline, entropy semantics, reuse settings, GP settings, inverse problem, bounded domain, observations/noise, and all other common settings remain fixed.

Use the same cumulative FOM budget across sweep cases. CI may use a reduced smoke grid, but the production sweep should retain FOM sizes `[4, 8, 16, 32]` and ROM-extra sizes `[64, 128]`.

Save per-case metadata including method, entropy strategy, FOM sample size, ROM-extra sample size, reuse enabled/disabled, seed, cumulative FOM evaluations, final parameter error, ELBO, and posterior summary. Write aggregate machine-readable results plus the resolved YAML.

## Fair-comparison requirements

Keep all common settings fixed unless intrinsically method-specific:

- identical prior and bounded parameter domain;
- identical truth and fixed noisy observations;
- identical observation covariance;
- identical initial variational distribution;
- LOO baseline throughout;
- identical algorithmic seed sets;
- identical FOM sample size where directly comparable;
- identical stopping/error criteria where meaningful;
- identical GP settings for all multifidelity runs.

For the two MFVI-Newton entropy runs, entropy strategy is the **only** intentional difference.

Use multiple independent algorithmic seeds for production results (target 10-20 if runtime permits), while holding the synthetic observation realization fixed.

## Primary cost metric

Use **cumulative FOM evaluations** as the primary x-axis. GP evaluations do not count as FOM evaluations. Record wall-clock time as a secondary diagnostic.

FOM accounting must include:

- ordinary BBVI samples;
- ARBIS refreshes;
- multifidelity HF samples;
- FOM samples used to construct/rebuild the automatic GP;
- any additional Newton/Hessian model evaluations.

## Outputs

At minimum generate:

1. parameter error vs cumulative FOM evaluations;
2. ELBO vs cumulative FOM evaluations;
3. per-parameter posterior-mean convergence with truth indicated;
4. final posterior mean and standard deviation/covariance;
5. total FOM evaluations and wall time;
6. median plus quantile bands for multi-seed runs;
7. machine-readable result data sufficient to regenerate plots without rerunning the benchmark.

The entropy ablation should have a direct comparison plot/table between `MF-BBVI (Newton, analytic entropy)` and `MF-BBVI (Newton, joint entropy)` using matched seeds and FOM work.

## Reuse diagnostics

For ARBIS variants save, where available:

- reuse vs refresh per iteration;
- refresh reason;
- archive/retained-sample count;
- ESS or equivalent importance-weight diagnostic;
- cumulative fresh FOM evaluations.

## CI / execution modes

Provide:

- a lightweight smoke configuration that executes all six primary run paths and checks finite outputs;
- a production configuration for scientific results;
- a lightweight CI sweep configuration that exercises the sweep machinery without running the full grid.

## Acceptance criteria

- [ ] Primary benchmark contains the five method families above, with MFVI-Newton split into analytic-entropy and joint-entropy runs (six production runs total).
- [ ] Both MFVI-Newton entropy runs use no sample reuse and differ only in `score_function_entropy_strategy`.
- [ ] Every applicable estimator uses the LOO baseline.
- [ ] All runs use the same bounded domain: `bmag in [0.5,1.5]`, `theta in [pi/6,pi/3]`, `nu in [0.02,0.08]`, `sigma in [0.1,0.6]`.
- [ ] Bounds are stored in YAML and enforced consistently.
- [ ] All multifidelity runs use the same automatic GP ROM; no coarse-grid LF model is used.
- [ ] Truth is off-center and strictly inside the bounded domain.
- [ ] Observation noise uses `sigma_obs = 0.05 * RMS(y_clean)` with a fixed observation seed.
- [ ] All runs and algorithmic seeds use the same fixed noisy dataset.
- [ ] Primary benchmark is YAML-driven through `configs/method_ablation.yaml` or equivalent.
- [ ] Sample-size sweep is YAML-driven through `configs/sample_size_sweep.yaml` or equivalent.
- [ ] Production sweep uses FOM sample sizes `[4, 8, 16, 32]` and MF ROM-extra sizes `[64, 128]` by default.
- [ ] Sweep includes both MFVI-Newton entropy strategies for every MF sample-size combination.
- [ ] Sweep cases share a fixed cumulative-FOM budget.
- [ ] Resolved YAML is copied into results.
- [ ] FOM accounting includes GP construction/rebuild and optimizer-specific FOM work.
- [ ] Main convergence figures use cumulative FOM evaluations.
- [ ] Results include parameter error, ELBO, posterior summaries, entropy strategy metadata, and machine-readable data.
- [ ] ARBIS runs expose reuse diagnostics.
- [ ] CI exercises all six primary paths and a reduced sweep.
- [ ] Production runs support repeated algorithmic seeds while holding observations fixed.

## Suggested labels

Use concise labels:

- `BBVI (Adam)`
- `ARBIS`
- `ARBIS + MF`
- `BBVI (Newton)`
- `MF-BBVI (Newton, analytic)`
- `MF-BBVI (Newton, joint)`

Suggested internal names:

- `bbvi_adam`
- `arbis`
- `arbis_mf`
- `bbvi_newton`
- `mf_bbvi_newton_analytic`
- `mf_bbvi_newton_joint`

**Comment by eparish1** [comment]
## Explicit bounded parameter-space requirement

Treat the CDR inverse problem as a **bounded four-dimensional parameter inference problem**. Use the bounded domain already used by the ROM-tools CDR UQ example:

- `bmag ∈ [0.5, 1.5]`
- `theta ∈ [π/6, π/3]`
- `nu ∈ [0.02, 0.08]`
- `sigma ∈ [0.1, 0.6]`

These bounds are part of the benchmark definition, not merely sampling ranges.

Requirements:

- all five methods must use exactly the same bounded parameter domain;
- the prior / initial variational distribution must be defined consistently with this domain;
- the synthetic truth must lie strictly inside the bounds and should remain off-center;
- the bounds must be specified explicitly in the benchmark YAML, not hard-coded only in Python;
- ARBIS, BBVI, and MF-BBVI paths must all enforce the same bounded-parameter handling / transform strategy;
- no method may gain an advantage by using clipping while another uses an unconstrained or transformed parameterization unless that difference is explicitly part of a separate ablation;
- saved result metadata should include the bounds and bounded-parameter strategy so the run is reproducible.

Add to the acceptance criteria:

- [ ] The CDR benchmark is explicitly defined on the bounded parameter domain above.
- [ ] All five methods enforce identical parameter bounds and the same bounded-parameter strategy.
- [ ] Bounds are specified in YAML and copied into the resolved run configuration/results.

**Comment by eparish1** [comment]
Implementation is now on branch `issue-393-cdr-vi-benchmark` (currently 10 commits ahead of `develop`). It adds the YAML-driven six-run CDR benchmark, production and smoke configurations, the FOM/ROM sample-size sweep, exact FOM accounting, fixed 5% RMS-noise observations, bounded transforms, ARBIS reuse diagnostics, the MFVI-Newton analytic-vs-joint entropy ablation, lightweight configuration tests, and a dedicated smoke workflow. No PR opened yet.

**Comment by eparish1** [comment]
This benchmark belongs in the VI/MFVI paper repository rather than ROMTools. The authoritative issue is now eparish1/mfvi-paper#5, with implementation on `issue-5-cdr-vi-benchmark`. Closing this ROMTools issue and removing the benchmark changes from the ROMTools implementation branch.
